src/parser/service/XmlParsingService.ts

```diff
@@ -0,0 +1,28 @@
+import { Builder, parseString } from 'xml2js';
+import * as Q from 'q';
+
+export default class XmlParsingService {
+  private static readonly xmlBuilder = new Builder({ headless: true });
+
+  public static toXml(obj: any): Q.Promise<string> {
+    const deferred = Q.defer<string>();
+    try {
+      deferred.resolve(this.xmlBuilder.buildObject(obj));
+    } catch (err) {
+      deferred.reject(err.message);
+    }
+    deferred.resolve();
+    return deferred.promise;
+  }
+
+  public static fromXml<T>(xml: string): Q.Promise<T> {
```

Comment:
Change this to return a native promise instead of a Q promise since it is a public function on a class that you are exporting in index.ts.